packages/react_rearch/lib/src/widgets/consumer.dart

```diff
@@ -0,0 +1,158 @@
+part of '../widgets.dart';
+
+abstract class RearchComponent extends Component2 {
```

Comment:
Funny that it is called `Component2`, but I looked quickly at the react package and that's just how they seem to do things.

Reply:
Yeah, `react` package is an exact map to React. There is another package `over_react` which is based on `react` and adds statically typing to the component props, but it uses a builder and add too much complexity. I just did today a custom mechanism to have a sort of typed props. But all of this goes outside this package.